pkg/util/log/log.go

```diff
@@ -38,7 +38,8 @@ func init() {
 // InitLogger initialises the global gokit logger (util_log.Logger) and overrides the
 // default logger for the server.
 func InitLogger(cfg *server.Config) {
-	l, err := NewPrometheusLogger(cfg.LogLevel, cfg.LogFormat)
+	factory := NewPrometheusLoggerFactory()
+	l, err := factory(cfg.LogLevel, cfg.LogFormat)
```

Comment:
Factory looks good, but i think we are gonna need to change the input for the function.
It really depend on what we are going to use, but i would probably send the server.Config and maybe logger.Config (if that is the approach we are taking), but for now i would change this to be passing the whole server.Config